list.go

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// This file is used to show the list of container checkpoints
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sync"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+func list(cmd *cobra.Command, args []string) error {
+	allPaths := append([]string{defaultCheckpointPath}, additionalCheckpointPaths...)
+
+	var wg sync.WaitGroup
+
+	table := tablewriter.NewWriter(os.Stdout)
+	header := []string{
+		"Namespace",
+		"Pod",
+		"Container",
+		"Time Created",
+		"Checkpoint Name",
+	}
+
+	table.SetHeader(header)
+	table.SetAutoMergeCells(false)
+	table.SetRowLine(true)
+
+	for _, checkpointPath := range allPaths {
+		fmt.Printf("Listing checkpoints in path: %s\n", checkpointPath)
+
+		files, err := filepath.Glob(filepath.Join(checkpointPath, "checkpoint-*"))
+		if err != nil {
+			return err
+		}
+
+		if len(files) == 0 {
+			fmt.Println("No checkpoints found in this path.")
+			continue
+		}
+
+		for _, file := range files {
+			wg.Add(1)
+
+			go func(file string) {
+				defer wg.Done()
+
+				namespace, pod, container, timestamp, err := parseCheckpointName(filepath.Base(file))
+				if err != nil {
+					log.Printf("Error parsing checkpoint name %s: %v\n", file, err)
+					return
+				}
+
+				timeCreated := timestamp
+
+				row := []string{
+					namespace,
+					pod,
+					container,
+					timeCreated,
+					filepath.Base(file),
+				}
+
+				table.Append(row)
+			}(file)
+		}
+	}
+
+	wg.Wait()
+
+	table.Render()
```

Comment:
It would be better to render a table only when there are any checkpoints found.
For example, something like the following:
```diff
 @@ -18,6 +18,7 @@ import (
 
 func list(cmd *cobra.Command, args []string) error {
 	allPaths := append([]string{defaultCheckpointPath}, additionalCheckpointPaths...)
+	showTable := false
 
 	var wg sync.WaitGroup
 
@@ -35,7 +36,6 @@ func list(cmd *cobra.Command, args []string) error {
 	table.SetRowLine(true)
 
 	for _, checkpointPath := range allPaths {
-		fmt.Printf("Listing checkpoints in path: %s\n", checkpointPath)
 
 		files, err := filepath.Glob(filepath.Join(checkpointPath, "checkpoint-*"))
 		if err != nil {
@@ -43,10 +43,11 @@ func list(cmd *cobra.Command, args []string) error {
 		}
 
 		if len(files) == 0 {
-			fmt.Println("No checkpoints found in this path.")
 			continue
 		}
 
+		showTable = true
+		fmt.Printf("Listing checkpoints in path: %s\n", checkpointPath)
 		for _, file := range files {
 			wg.Add(1)
 
@@ -76,7 +77,11 @@ func list(cmd *cobra.Command, args []string) error {
 
 	wg.Wait()
 
-	table.Render()
+	if showTable {
+		table.Render()
+	} else {
+		fmt.Println("No checkpoints found")
+	}
 
 	return nil
 }
```